iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp

```diff
@@ -321,6 +321,36 @@ class Timer
 };
 
 } // namespace posix
+
+// IOX-#337 Alternate Timer to replace the POSIX timer
```

Comment:
or maybe even better make an `@brief` with a small description. In one year nobody cares that this functionality was in POSIX Timer

Reply:
Can you make the class comment more descriptive ?
What does it mean to be a deadline timer ? How do I use it ?

Reply:
Okay added more descriptions for the class

Reply:
Also, this file should be renamed to deadline_timer.hpp

Reply:
no, the functionality needs to be moved to a new file. This file will contain a new timer which just wraps a semaphore and does a blocking wait until it's either interrupted or timed out. It will be used in `PeridoicTask`, which in combination with the new timer provides the functionality of the execution of a callback from the old timer.
The semaphore might later on be replaced by a `timer_create` call, but instead of spawning a thread, it will just block, like the semaphore and if someone needs a separate thread, the `PeriodicTask` should be used.